### ISSUE TITLE
Allow multiple prefixes in the Feather naming rules

### ISSUE THREAD

**Issue by iampremo** [YoYoGames/GameMaker-Bugs#4110]
Requested in ticket Ticket #202365

"Description: I would like an ability to define multiple alternative prefixes for a given resource type naming rules. For example, "spr_" and "bg_" for sprites, instead of only one or the other.
Right now, if I try to enter a prefix like "spr_, bg_" I instead get a naming rule suggestion to use a "spr_, bg_SpriteName" name instead of spr_SpriteName, for example.

The alternative prefixes could be given in a comma-separated list - since commas can't be a part of a valid identifier anyway, they can't be mistaken for the actual part of the prefix. So "spr_, bg_" combined with UpperCamelCase core would allow names like "spr_SomeSprite" and "bg_SomeSprite", but not "obj_SomeSprite" or plain "SomeSprite".

When it comes to the GM2017 rule message, I see two options of adapting it:
1. Recommend the first of the given prefixes; e.g. given an alternative prefixes of "spr_, bg_" it would show a message like: "Inconsistent naming. Recommended name is 'spr_SomeSprite'."
2. List all the alternatives; e.g. given an alternative prefixes of "spr_, bg_" it would show a message like: "Inconsistent naming. Recommended names are: 'spr_SomeSprite', 'bg_SomeSprite'."
Benefit: In my projects, I often use multiple types of prefixes for a specific asset type. For example, I would use:
- spr_Wall and bg_Sky for sprite assets
- bgm_Forest and sfx_Thud for audio assets
- obj_Player, ctrl_Overworld and ui_LevelTitle for object assets

At the moment I can apply only one prefix in my naming rules, and dropping the prefix altogether means that I can't mix the underscore in the prefix with UpperCamelCase for the main name. So I can't really use the system to enforce my naming conventions - ability to define multiple alternative prefixes would fix it."

**Comment by noahNcopeland** [comment]
oh baby I need this so bad. Surprised its not already a thing, given how common it is to have separate prefixes for music and sounds, and also objects and parents. 

My standards for audio are very organized, but feather doesn't like it. 
musMusic
ambAmbience
folFoley
sndSound
vocVoice

**Comment by Notlea** [comment]
I came here to make a feature request and turns out it already exists. I was surprised it isn't a thing though.

**Comment by Feronar** [comment]
Yes, please.  I prefer to use multiple prefixes for different uses of my audio assets, i.e snAsset, vcEnAsset, muAsset, etc.  I also use obAsset for ordinary objects and ccAsset for my core controller objects (Like ccSystem for controlling core software functionality, ccJourney for story and character progression, ccInput for keyboard/gamepad, etc).
